components/renku_data_services/project/core.py

```diff
@@ -95,6 +96,15 @@ async def copy_project(
     )
     project = await project_repo.insert_project(user, unsaved_project)
 
+    # NOTE: Copy session secret slots
+    secret_slots = await session_secret_repo.get_all_session_secret_slots_from_project(
+        user=user, project_id=source_project_id
+    )
+    for secret_slot in secret_slots:
+        await session_secret_repo.copy_session_secret_slot(
+            user=user, project_id=project.id, session_secret_slot=secret_slot
+        )
+
```

Comment:
Could this be done in the db entirely maybe? with `insert … select …` or something? not blocking, just wondering. could be quicker and perhaps easier to write as well.